nilearn/regions/tests/test_hierarchical_kmeans_clustering.py

```diff
@@ -151,16 +160,17 @@ def test_hierarchical_k_means_clustering_surface(
     # create a surface masker
     masker = SurfaceMasker(surf_mask()).fit()
     # mask the surface image with 50 samples
-    X = masker.transform(surf_img_2d(50)).T
+    X = masker.transform(surf_img_2d(100))
     # instantiate HierarchicalKMeans with n_clusters
-    clustering = HierarchicalKMeans(n_clusters=n_clusters)
+    hkmeans = HierarchicalKMeans(n_clusters=n_clusters)
     # fit and transform the data
-    X_transformed = clustering.fit_transform(X)
+    X_transformed = hkmeans.fit_transform(X)
     # inverse transform the transformed data
-    X_inverse = clustering.inverse_transform(X_transformed)
+    X_inverse = hkmeans.inverse_transform(X_transformed)
 
     # make sure the n_features in transformed data were reduced to n_clusters
     assert X_transformed.shape[0] == n_clusters
```

Comment:
I think the transform should reduce the number of features because, in the end, we want to have one feature that represents many similar features.

See the docs:
![Screenshot 2024-12-12 at 10 46 20 AM](https://github.com/user-attachments/assets/8f941d1c-9c53-43f7-b8f5-b71cc4f4dede)

Reply:
yeah but I added a check for this and it fails

https://github.com/nilearn/nilearn/actions/runs/12293225132/job/34305525450?pr=4939#step:6:27061

```
FAILED nilearn/regions/tests/test_hierarchical_kmeans_clustering.py::test_hierarchical_k_means_clustering - assert 7 == 8
```

Reply:
why does it do that?

sorry (task switching means I can't follow this one)